services/scan/src/precinct_scanner_app.ts

```diff
@@ -551,9 +598,21 @@ export async function buildPrecinctScannerApp(
     }
   );
 
-  app.post<NoParams, OkResponse>(
+  app.post<NoParams, OkResponse | ErrorsResponse>(
     '/precinct-scanner/scanner/scan',
     (_request, response) => {
+      if (store.getPollsState() !== 'polls_open') {
+        response.status(400).json({
+          status: 'error',
+          errors: [
+            {
+              type: 'polls-closed',
+              message: 'cannot scan ballots while polls are closed',
+            },
+          ],
+        });
+        return;
+      }
```

Comment:
I'm not sure if this is too limiting - to disabled scanning when polls are closed. Any reservations about this? Seems like a security in depth way to lock down "no voting while polls closed" requirements.

Reply:
I like this